### DISCUSSION TITLE
Version 1.0.0 behind Traefik reverse proxy does not serve static files

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox-docker#437] [Getting Help]
Trying to get the new version running behind Traefik on Docker, where i use http://server/ipam/ url for Netbox

Now, in earlier versions, I had a nginx.conf with adoption to the BASE_PATH=ipam/ in Netbox ENV

Since nginx is removed i favor for nginx-unit, i suspect there is some config issue on my side. Anyone that have done 1.0 behind Traefik?

**Comment by ziggekatten** [comment]
And I found oout what is happening. 

The section in nginx-unit.json appends the BASE_PATH to the share statement. Using BASE_PATH=netbox and a nginx-unit.json config with below works:

  "routes": [
    {
      "match": {
        "uri": "/netbox/static/*"
      },
      "action": {
        "share": "/opt/netbox"
      }
    }

How on earth do i get nginx unit not to behave like that? We use various URLs and base paths http://server/ipam-dev, http://server/ipam-test ,http://server/ipam-prod) etc, and was very simple with the old nginx.conf as a docker config for each instance. Cant find to set Unit to use a specific directory

**Comment by ziggekatten** [comment]
FYI, I ended up with adding a Traefik rule n my deployment, so that I still use ngninx and a shared static folder. Only solution I found to the issue

**Comment by TheNetworkGuy** [comment]
FYI i'm struggling with the exact same issue. Running netbox-docker behind a reverse proxy with URL test123.org.local/netbox. Setting the BASE_PATH does not work and results in a static media failure.

After manually adjusting the nginx-unit.json file in the container i'm unable to fix the situation. I guess i'll revert back to 0.27 but this is a real shame.

If any of the maintainers need extra information in regards to configuration or logs etc i would happy to assist. I'm not that confident with nginx-unit files so its a bit difficult for me to make a final solution.

**Reply by ziggekatten** [reply]
I removed the static serving from unit, added a traditional ngninx container for static serving, and mount a shared nfs resource for static files in the netbox containers. 

So, I still use unit on the netbox containers, since it's WAY more stable than earlier implementation with gunicorn. Especially under load where api calls would give erratic 500 errors, we no longer have these issues.

**Comment by 7oku** [comment]
We actually just faced the same issue and found a pretty simple solution: 

In `docker-compose.yml` disable traefik for the netbox worker:

```
  netbox-worker:
    <<: *netbox
    depends_on:
    - redis
    entrypoint:
    - /opt/netbox/venv/bin/python
    - /opt/netbox/netbox/manage.py
    command:
    - rqworker
    ports: []
    deploy:
      labels:
        - traefik.enable=false
```

After nginx is removed you have 2 endpoints which traefik is loadbalancing to: the netbox container and the netbox-worker container. These both however are constructed from a template and your traefik rules are applied to both of them resulting in both of them to be targeted by traefik, while only the netbox container serves the files. 

Disabling trafik and thus overriding the templated rules directs all traffic to the netbox container and things work like expected.

**Reply by Yoyasp** [reply]
This also worked for me, altough i had to change a bit of syntax:
`
  netbox-worker:
    <<: *netbox
    depends_on:
    - redis
    entrypoint:
    - /opt/netbox/venv/bin/python
    - /opt/netbox/netbox/manage.py
    command:
    - rqworker
    ports: []
    labels:
        - "traefik.enable=false"`

I also added those last lines to the housekeeping container

**Reply by s0p4L1n3** [reply]
thank you @Yoyasp  ! it helped me years later (netbox 3.7.2-2.8.0) with traefik 2.11 as the problem still exist...

**Reply by rock7632** [reply]
I tried doing this but can't get it to work. still getting bad gateway when I go to the url. Can you tell me if your traefik labels are in the docker-compose.yml or the docker-compose-override.yml and also did you have put anything in the housekeeping container?

**Reply by s0p4L1n3** [reply]
@rock7632 I’ll give you my full docker compose file on Monday when I get back to my computer 👍

**Reply by rock7632** [reply]
@s0p4L1n3 Thanks

**Reply by s0p4L1n3** [reply]
@rock7632 https://github.com/s0p4L1n3/netbox-docker-traefik-TLS/blob/main/docker-compose.yml

**Reply by rock7632** [reply]
Thanks. Let me try it out

**Reply by rock7632** [reply]
@s0p4L1n3 This worked for me. Thank you so much!

**Comment by ziggekatten** [comment]
Did anyone succeed with this? The suggestion to disable traefik did nothing. I never got it working, and with the latest 1.4 version I'm unable to use a shared volume for static files as the required user 101 have no write permissions on my NFS volume. This Unit stuff stinks big time

**Reply by kale1d0code** [reply]
I've been trying a few things to get netbox to work with traefik

as well as the above suggestion for disabling traefik on the worker
I also tried using the stripprefix middleware which removes the base_path before sending it onto the netbox container

```

services:
  netbox:
  ...
  labels:
  ...
  - 'traefik.http.middlewares.<MIDDLEWARE_NAME>.stripprefix.prefixes=/netbox'
  - 'traefik.http.routers.netbox.middlewares=<MIDDLEWARE_NAME>@docker,...'
  ...
```

this fixes the static file warning and I can get to the UI
but this then causes any part of the app you try to go to serve a netbox 404 page.

hoping this is something to go on

next thing I'm going to try is seeing if I can get the middleware only to run on static assets

**Comment by sfulham** [comment]
I'm also getting the same issue, but I'm proxying it with a subdomain, rather than a path, and I get the same issue. Although, according to the network tab in Chrome, two of the static assets load, and then the rest timeout with a 504. Any ideas?

**Comment by courtland** [comment]
I ran into this as well today. The problem is that the django application is aware of `BASE_PATH ` however the static assets and nginx-unit are not. I worked around this by using a `replacepathregex` rule in Traefik in addition to the Router rule that has the `PathPrefix`. For example, replacing `/netbox/static` with `/static` ensures that nginx-unit routes static asset requests correctly, and the app URL/routes are left alone.

I would think changing nginx-unit config to include the desired path prefix in the static route would also work.

**Comment by courtland** [comment]
FWIW this is also discussed in https://github.com/netbox-community/netbox-docker/issues/650 with  similar resolution / workaround options.
